compiler/src/dotty/tools/dotc/typer/ErrorReporting.scala

```diff
@@ -98,7 +98,13 @@ object ErrorReporting {
       val normTp = normalize(tree.tpe, pt)
       val treeTp = if (normTp <:< pt) tree.tpe else normTp
         // use normalized type if that also shows an error, original type otherwise
-      errorTree(tree, TypeMismatch(treeTp, pt, implicitFailure.whyNoConversion))
+      def missingElse = tree match
+        case If(_, _, elsep @ Literal(Constant(()))) if elsep.span.isSynthetic =>
+          "\nMaybe you are missing an else part for the conditional?"
```

Comment:
If the Unit in the error message is the confusing part, maybe we could explain that:
```suggestion
          "\nNote: an `if` without a corresponding `else` will always have type `Unit`"
```
Not sure if that's better.

Reply:
I the single line is enough of a hint. An intelligent reader will connect the dots.